### ISSUE TITLE
Soft deletion still has issue

### ISSUE THREAD

**Issue by rifqi-ridwan** [go-rel/rel#227]
Hi, i tried to use soft deletion to see if it's work  
but somehow i found this still has a bug especially on mysql and mariadb  

here some example while inserting on mysql in docker  
![image](https://user-images.githubusercontent.com/29592413/137128041-12ca7a52-a60e-41ae-886a-96f8b4c0a212.png)

it's inserted '0000-00-00' value but querying with `deleted_at IS NULL`, so basically it will always return empty since '0000-00-00' is considered as not null value  
i knew we could avoid this using `rel.Unscoped(true)`, but this will also break autoload query
i tried 2 scenario using association with 2 table, 1st product and the 2nd is category  using 1-1

scenario 1:  
product table without deleted_at, category table using deleted_at, like i said before the autoload would return empty category while querying from product(IS NULL problem while querying category)
scenario 2:
product and category table had deleted_at field, i couldn't get the product except i'm using unscoped, but if i'm using it, the autoload doesn't work

here the example on mariadb in xampp  
![image](https://user-images.githubusercontent.com/29592413/137129198-f5a14ee8-b55b-43ec-847b-d77dfa43b6f0.png)

on mysql insert still had a problem i'm not sure why it happened, but it's different on mariadb the insert still work perfectly fine  
so basically to make this work maybe we should change the query to `SELECT * FROM 'categories' WHERE deleted_at = '0000-00-00 00:00:00';`  
or we could just add `OR` at `where` condition

**Comment by Fs02** [comment]
I see, no worries 😉 

DeletedAt needs to be pointer, otherwise it won’t be able to save or store null value

**Comment by rifqi-ridwan** [comment]
i'm really2 sorry about this  
i guess it's my bad  

please consider to mention it here
![image](https://user-images.githubusercontent.com/29592413/137166647-979a49f8-65b2-48c5-b66b-d0a638b2cba7.png)

thank you for answering this stupid question
and thank you so much for this awesome module
